crates/ide_completion/src/context.rs

```diff
@@ -744,8 +744,8 @@ impl<'a> CompletionContext<'a> {
         // Overwrite the path kind for derives
         if let Some((original_file, file_with_fake_ident, offset)) = derive_ctx {
             let attr = self
-                .sema
-                .token_ancestors_with_macros(self.token.clone())
```

Comment:
So this works in the common case, but this breaks down as soon as the `derive` is inside a `macro_call!(...)`, rare but still something we should be able to support somewhat I ***think***. So what we would need to do here is to specifically map out of the "pseudo-derive expansion", and then look for the attribute.

So I think what we should do here is when doing the pseudo derive expansion in `expand_and_fill` (the part with the `// Expand pseudo-derive expansion` comment), note down the `orig_attr` there in the `derive_ctx`, and use that instead of searching for the attribute like we do here. The search is unnecessary since we already know the derive attribute from that part of the code, as `expand_derive_as_pseudo_attr_macro` succeeding already verifies that this is in fact a derive attribute.

Reply:
Thanks for the guide, I think I get what you mean, I'll try to revise it.

Reply:
Feel free to ask if something's unclear

Reply:
I'm done, you’ve made it so clear, I just follow what you said step by step.